### ISSUE TITLE
Limit the number of HTTP connection using outputs.wavefront

### ISSUE THREAD

**Issue by crflanigan** [influxdata/telegraf#14090]
### Use Case

Please add a feature request for the outputs.wavefront Telegraf output plugin which can control for the number of connections the plugin makes such that we can pin it to a specific value, like outputs.http does now.
This is to avoid the Wavefront plugin from making too many connection and overloading the proxies.

### Expected behavior

We set the number of HTTP connections (like outputs.http does now) to a certain value and it stays that way.

Example from outputs.http:

```
  ## MaxIdleConns controls the maximum number of idle (keep-alive)
  ## connections across all hosts. Zero means no limit.
  # max_idle_conn = 0

  ## MaxIdleConnsPerHost, if non-zero, controls the maximum idle
  ## (keep-alive) connections to keep per-host. If zero,
  ## DefaultMaxIdleConnsPerHost is used(2).
  # max_idle_conn_per_host = 2

  ## Idle (keep-alive) connection timeout.
  ## Maximum amount of time before idle connection is closed.
  ## Zero means no limit.
  # idle_conn_timeout = 0
```

### Actual behavior

This doesn't currently exist.

**Comment by powersj** [comment]
@crflanigan,

The wavefront output uses the [wavefront-sdk-go](https://github.com/wavefrontHQ/wavefront-sdk-go) client to manage the connection and sending of metrics.

@LukeWinikates,

Are these types of options available with the wavefront client or are there better ways for Telegraf to generate the client to not cause issues when using a proxy?

Thanks!

**Comment by LukeWinikates** [comment]
Hi friends,

I spent some time looking into this issue this week (we have a Wavefront support thread open for @crflanigan's case).

As written, the Telegraf wavefront output only ever sends 1 http request at a time to the Wavefront proxy or Wavefront host. It's basically a ticker that makes a single `POST` request to `url` on each tick. Proxies from our test environments all show 1 connection at a time from each Telegraf output instance, and even when we artificially induce HTTP timeouts, we don't see any leaking of open connections.

That said, we've only tested on Mac and Linux, not Windows

We can take a look at adding:

```
	IdleConnTimeout     config.Duration `toml:"idle_conn_timeout"`
	MaxIdleConns        int             `toml:"max_idle_conn"`
	MaxIdleConnsPerHost int             `toml:"max_idle_conn_per_host"`
```

and passing those to the go HTTP client the SDK uses for its post requests. I don't see any harm in doing that, so we'll give it a try next week in hopes of getting it into the next Telegraf release.

**Comment by crflanigan** [comment]
@LukeWinikates 

What version of the agents and proxies were you using for testing.

**Comment by LukeWinikates** [comment]
@crflanigan we were using Telegraf v1.28.2, and v13 of the Wavefront proxy. For some of our tests, we didn't use the Wavefront proxy - we created a small "fake proxy" with an an 11s sleep on every request, and used that to cause timeouts with the http requests from Telegraf/the Wavefront output plugin, so we could observe whether Telegraf failed to close its server connections.

I wouldn't say that we proved that the Wavefront output plugin never leaks server connections, but we weren't able to create a situation where it did. Maybe there's another variable we haven't identified yet. I do like the suggestion of adding the parameters that you identified!
